### PR TITLE
Add subcommand to fetch a deployment manifest for an edge region

### PR DESCRIPTION
**Pull request by lukasbals**
## Summary of the changes / Why this is an improvement

Add subcommand to fetch a deployment manifest for an edge region.

On hold since the backend endpoint is not yet released.

## Checklist

 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

File: croud/regions/commands.py [hunk 1]
```diff
@@ -37,3 +38,28 @@ def regions_list(args: Namespace) -> None:
...
+    """
+
+    client = Client.from_args(args)
+    print_info("Generating deployment manifest ...")
+    data, errors = client.get(
+        f"/api/v2/regions/{args.region_name}/deployment-manifest/"
+    )
+    if data:
+        try:
+            f = open(data["name"], "x")
```
Comment (SStorm) [inline croud/regions/commands.py:56]: Whoa, totally not obvious that it's going to write to a file here... Would it be better to print this to `stdout` and let the user of `croud` redirect it where he wants to?

Comment (SStorm) [reply croud/regions/commands.py:56]: I mean, what if the server returns '/etc/passwd' and I'm running croud as root? :D

Comment (lukasbals) [reply croud/regions/commands.py:56]: Good point. Working on a solution.

File: croud/regions/commands.py [hunk 2]
```diff
@@ -37,3 +38,32 @@ def regions_list(args: Namespace) -> None:
...
+
+    client = Client.from_args(args)
+    data, errors = client.get(
+        f"/api/v2/regions/{args.region_name}/deployment-manifest/"
+    )
+    if data:
+        content = b64decode(data["content"]).decode()
+        if args.file_name:
+            try:
+                f = open(args.file_name, "x")
```
Comment (SStorm) [inline croud/regions/commands.py:57]: You could use `with open(...) as file:` I suppose :shrug:

**Review by SStorm — APPROVED** [review]
LGTM!

File: croud/regions/commands.py [hunk 3]
```diff
@@ -37,3 +38,32 @@ def regions_list(args: Namespace) -> None:
...
+    data, errors = client.get(
+        f"/api/v2/regions/{args.region_name}/deployment-manifest/"
+    )
+    if data:
+        content = b64decode(data["content"]).decode()
+        if args.file_name:
+            try:
+                with open(args.file_name, "x") as file:
+                    file.write(content)
+                    file.close()
```
Comment (SStorm) [inline croud/regions/commands.py:59]: You don't need `file.close()` if you use `with ...` :stuck_out_tongue:

**Review by matthijskrul — APPROVED** [review]

**Comment by lukasbals** [comment]
I will close this PR since there has a feature branch been created: https://github.com/crate/croud/pull/316

I will create a new PR against the feature branch: `feature/bootstrap-edge`

**Pull request closed without merging**